Build fails under Java 9 due to java.lang.StringIndexOutOfBoundsException in maven-javadoc-plugin

When trying to build under Java 8 I get the following error:

```
[INFO] --- maven-javadoc-plugin:2.10.3:javadoc (generate-javadoc) @ dkpro-jwktl ---
[WARNING] Error injecting: org.apache.maven.plugin.javadoc.JavadocReport
java.lang.ExceptionInInitializerError
	at org.apache.maven.plugin.javadoc.AbstractJavadocMojo.<clinit>(AbstractJavadocMojo.java:195)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:86)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:105)
	at com.google.inject.internal.ConstructorInjector.access$000(ConstructorInjector.java:32)
	at com.google.inject.internal.ConstructorInjector$1.call(ConstructorInjector.java:89)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:115)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:133)
	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:68)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:87)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at org.eclipse.sisu.space.AbstractDeferredClass.get(AbstractDeferredClass.java:48)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision(InternalFactoryToInitializableAdapter.java:53)
	at com.google.inject.internal.ProviderInternalFactory$1.call(ProviderInternalFactory.java:65)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:115)
	at org.eclipse.sisu.bean.BeanScheduler$Activator.onProvision(BeanScheduler.java:176)
	at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:126)
	at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:68)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:63)
	at com.google.inject.internal.InternalFactoryToInitializableAdapter.get(InternalFactoryToInitializableAdapter.java:45)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	at org.eclipse.sisu.inject.Guice4$1.get(Guice4.java:162)
	at org.eclipse.sisu.inject.LazyBeanEntry.getValue(LazyBeanEntry.java:81)
	at org.eclipse.sisu.plexus.LazyPlexusBean.getValue(LazyPlexusBean.java:51)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:263)
	at org.codehaus.plexus.DefaultPlexusContainer.lookup(DefaultPlexusContainer.java:255)
	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo(DefaultMavenPluginManager.java:517)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:121)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 1
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3116)
	at java.base/java.lang.String.substring(String.java:1885)
	at org.apache.commons.lang.SystemUtils.getJavaVersionAsFloat(SystemUtils.java:1133)
	at org.apache.commons.lang.SystemUtils.<clinit>(SystemUtils.java:818)
	... 59 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```
This happens with `maven-javadoc-plugin` version 2.10.3. Same story with `2.10.4`.
Upgrading to `3.0.1` solves this issue.

--- a/pom.xml
+++ b/pom.xml
@@ -138,6 +138,12 @@
         <version>3.0.1</version>
         <configuration>
           <additionalparam>${javadoc.opts}</additionalparam>
+          <additionalJOptions>
+            <additionalJOption>-Xmaxerrs</additionalJOption>
+            <additionalJOption>1000</additionalJOption>
+            <additionalJOption>-Xmaxwarns</additionalJOption>
+            <additionalJOption>1000</additionalJOption>
+          </additionalJOptions>
         </configuration>
         <executions>
           <execution>

--- a/pom.xml
+++ b/pom.xml
@@ -135,6 +135,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-javadoc-plugin</artifactId>
+        <version>3.0.1</version>
         <configuration>
           <additionalparam>${javadoc.opts}</additionalparam>
         </configuration>
@@ -152,7 +153,15 @@
 
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.0</version>
+      </plugin>
+
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-assembly-plugin</artifactId>
+        <version>3.1.0</version>
         <executions>
           <execution>
             <phase>package</phase>

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/api/entry/WikiString.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/api/entry/WikiString.java
@@ -54,7 +54,7 @@ public class WikiString implements IWikiString {
 		this.text = text;
 	}
 	
-	/** @see #getText(). */
+	/** @see #getText() */
 	public String getTextIncludingWikiMarkup() {
 		return getText();
 	}

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/api/util/TemplateParser.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/api/util/TemplateParser.java
@@ -76,13 +76,13 @@ public /*static*/ class TemplateParser {
 				namedParams.put(key, value);
 		}
 
-		/** Return the numbered parameter with the given index or <code>null/<code> 
+		/** Return the numbered parameter with the given index or <code>null</code> 
 		 *  if no such parameter exists. */
 		public String getNumberedParam(int idx) {
 			return numberedParams.get(idx);
 		}
 		
-		/** Return the named parameter with the given key or <code>null/<code> 
+		/** Return the named parameter with the given key or <code>null</code> 
 		 *  if no such parameter exists. */
 		public String getNamedParam(final String key) {
 			return namedParams.get(key);

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/en/components/ENSemanticRelationHandler.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/en/components/ENSemanticRelationHandler.java
@@ -30,7 +30,7 @@ import static de.tudarmstadt.ukp.jwktl.api.entry.WikiString.removeWikiLinks;
  * Parser component for extracting
  * <a href="https://en.wiktionary.org/wiki/Wiktionary:Semantic_relations">semantic relations</a>
  * from the English Wiktionary.
- * <p/>
+ * <p>
  * A semantic relation ship can specify a target sense. This handler tries to determine the target
  * sense using {@link #findMatchingSense(WiktionaryEntry, String)}.
  */

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/language/WikimediaSisterProject.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/language/WikimediaSisterProject.java
@@ -109,7 +109,7 @@ public class WikimediaSisterProject {
 
     
     /** Gets texts of interwikimedia link. E.g. 
-     * [[wikt:Wiktionary:Statistics#Detail|statistics]]  -> "statistics"
+     * [[wikt:Wiktionary:Statistics#Detail|statistics]]  -&gt; "statistics"
      * 
      * @param text_inside_link  e.g. "wikipedia:" in [[wikipedia:]]
      */
@@ -127,7 +127,7 @@ public class WikimediaSisterProject {
     }
     
     /** Gets texts of interwikimedia link. E.g. 
-     * [[wikt:Wiktionary:Statistics#Detail|statistics]]  -> "statistics"
+     * [[wikt:Wiktionary:Statistics#Detail|statistics]]  -&gt; "statistics"
      * 
      * @param before the text before the first column, e.g. "wikipedia" in [[wikipedia:]]
      * @param after e.g. "Wikipedia:Wikimedia_sister_projects" in [[w:Wikipedia:Wikimedia_sister_projects]]

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/ReferenceParser.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/ReferenceParser.java
@@ -20,7 +20,7 @@ import java.util.regex.Pattern;
 
 import de.tudarmstadt.ukp.jwktl.parser.ru.wikokit.base.wikipedia.util.StringUtil;
 
-/** Parser of wiki references &lt;ref>...&lt;/ref>
+/** Parser of wiki references &lt;ref&gt;...&lt;/ref&gt;
  */
 public class ReferenceParser {
     
@@ -49,7 +49,7 @@ public class ReferenceParser {
     
     /** Expands texts of the refence, and adds it to the end of text.
      * 
-     * If the reference contains a template, e.g. &lt;ref>{{cite book |..&lt;/ref>
+     * If the reference contains a template, e.g. &lt;ref&gt;{{cite book |..&lt;/ref&gt;
      * then the whole reference will be deleted.
      */ 
     public static StringBuffer expandMoveToEndOfText(StringBuffer text)

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/WikiParser.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/WikiParser.java
@@ -44,9 +44,9 @@ public class WikiParser {
     private final static Pattern ptrn_pre_code     = Pattern.compile("<pre>.+?</pre>",      Pattern.DOTALL);
     private final static Pattern ptrn_source_code  = Pattern.compile("<source.+?</source>", Pattern.DOTALL);
     
-    // 1. simple wikilink without '|' inside link, e.g. [[Tsar]] -> Tsar
+    // 1. simple wikilink without '|' inside link, e.g. [[Tsar]] -&gt; Tsar
     private final static Pattern ptrn_remove_brackets_in_wikilinks = Pattern.compile("\\[\\[([^:|]+?)\\]\\]");
-    // 2. with '|' inside link, e.g. [[The Russian language|Russian]] -> Russian
+    // 2. with '|' inside link, e.g. [[The Russian language|Russian]] -&gt; Russian
     // [^[:|]| - not '[', ':' till first '|'
     //private final static Pattern ptrn_remove_brackets_in_wikilinks_vertical_line = Pattern.compile("\\[\\[[^[:]*?|(.+?)\\]\\]");
     //private final static Pattern ptrn_remove_brackets_in_wikilinks_vertical_line = Pattern.compile("\\[\\[[^\Q:]\E]*?|(.+?)\\]\\]");
@@ -77,7 +77,7 @@ public class WikiParser {
     /** Creates a new instance of WikiParser */
     //public WikiParser() {    }
     
-    /** Removes interwiki, e.g. "[[et:Talvepalee]] text" -> " text", 
+    /** Removes interwiki, e.g. "[[et:Talvepalee]] text" -&gt; " text", 
      * where language code (e.g. 'et') can have two or three letters.
      */
     public static StringBuffer removeInterwiki(StringBuffer text) {
@@ -86,7 +86,7 @@ public class WikiParser {
     }
     
     /** Expands interwiki by removing interwiki brackets and language code, 
-     * e.g. "[[et:Talvepalee]] text" -> "Talvepalee text".
+     * e.g. "[[et:Talvepalee]] text" -&gt; "Talvepalee text".
      */
     public static StringBuffer removeBracketsInInterwiki(StringBuffer text) {
         Matcher m = ptrn_remove_brackets_in_interwiki.matcher(text.toString());
@@ -102,8 +102,8 @@ public class WikiParser {
     }
     
     /** Removes categories for selected language, 
-     * e.g. English: "[[Category:Russia]] text" -> " text", 
-     * or Esperanto: "[[Kategorio:Galaksioj]] text" -> " text".
+     * e.g. English: "[[Category:Russia]] text" -&gt; " text", 
+     * or Esperanto: "[[Kategorio:Galaksioj]] text" -&gt; " text".
      */
     public static StringBuffer removeCategory(StringBuffer text, LanguageType lang) {
         Matcher m = null;
@@ -129,25 +129,25 @@ public class WikiParser {
     }
     
     /** Removes XML tag <code> with text till the next </code>.
-     * e.g. "a <code>x+y</code> b" -> "a  b". */
+     * e.g. "a <code>x+y</code> b" -&gt; "a  b". */
     public static StringBuffer removeXMLTagCode(StringBuffer text) {
         Matcher m = ptrn_tag_code.matcher(text.toString());
         return new StringBuffer(m.replaceAll(""));
     }
     
-    /** Removes all comments: &lt;!-- ... -->. */
+    /** Removes all comments: &lt;!-- ... --&gt;. */
     public static StringBuffer removeHTMLComments(StringBuffer text) {
         Matcher m = ptrn_html_comment.matcher(text.toString());
         return new StringBuffer(m.replaceAll(""));
     }
     
-    /** Removes preformatted code (e.g. xml): &lt;pre> ... &lt;/pre>.*/
+    /** Removes preformatted code (e.g. xml): &lt;pre&gt; ... &lt;/pre&gt;.*/
     public static StringBuffer removePreCode(StringBuffer text) {
         Matcher m = ptrn_pre_code.matcher(text.toString());
         return new StringBuffer(m.replaceAll(""));
     }
     
-    /** Removes all source codes: &lt;source ... &lt;/source>.*/
+    /** Removes all source codes: &lt;source&gt; ... &lt;/source&gt;.*/
     public static StringBuffer removeSourceCode(StringBuffer text) {
         Matcher m = ptrn_source_code.matcher(text.toString());
         return new StringBuffer(m.replaceAll(""));
@@ -155,8 +155,8 @@ public class WikiParser {
     
     
     /** Expands wiki links removing brackets. There are two cases: 
-     * (1) remove brackets, e.g. [[run]] -> run and 
-     * (2) (todo) [[run|running]] -> run, or [[Russian language|Russian] -> Russian, 
+     * (1) remove brackets, e.g. [[run]] -&gt; run and 
+     * (2) (todo) [[run|running]] -&gt; run, or [[Russian language|Russian] -&gt; Russian, 
      * i.e. the visible (to reader) words will remain.
      */
 	@Deprecated
@@ -186,7 +186,7 @@ public class WikiParser {
     }
     
     /** Expands / removes hyperlinks. Expands hyperlinks with text, e.g. 
-     * "[http:site name of site]" -> "name of site". 
+     * "[http:site name of site]" -&gt; "name of site". 
      * Removes links without text, e.g. [www.site].
      */
     public static StringBuffer parseSingleBrackets(StringBuffer text)
@@ -227,25 +227,25 @@ public class WikiParser {
     /** Removes and expands interwiki, categories, and wiki links in wiki texts.<br>
      * 
      * 1. expands links to Wikimedia sister projects, 
-     * see [[w:Wikipedia:Interwikimedia_links|text to expand]] -> "text to expand"
+     * see [[w:Wikipedia:Interwikimedia_links|text to expand]] -&gt; "text to expand"
      * 
      * 2. interwiki
      * @param b_remove_not_expand_iwiki if true then 
-     * Removes interwiki, e.g. "[[et:Talvepalee]] text" -> " text";<br>
+     * Removes interwiki, e.g. "[[et:Talvepalee]] text" -&gt; " text";<br>
      *                                  if false then
      * expands interwiki by removing interwiki brackets and language code,
-     * e.g. "[[et:Talvepalee]] text" -> "Talvepalee text".
+     * e.g. "[[et:Talvepalee]] text" -&gt; "Talvepalee text".
      *
      * @param  lang defines parsed wiki language, it is needed to remove 
      * category for the selected language, e.g. English (Category) or Esperanto 
      * (Kategorio).<br><br>
      *
      * 3. Removes categories for selected language, 
-     * e.g. English: "[[Category:Russia]] text" -> " text".<br><br>
+     * e.g. English: "[[Category:Russia]] text" -&gt; " text".<br><br>
      *
      * 4. Expands wiki links by removing brackets. There are two cases: 
-     * (1) remove brackets, e.g. [[run]] -> run and 
-     * (2) [[run|running]] -> running, or [[Russian language|Russian]] -> Russian, 
+     * (1) remove brackets, e.g. [[run]] -&gt; run and 
+     * (2) [[run|running]] -&gt; running, or [[Russian language|Russian]] -&gt; Russian, 
      * i.e. the visible (to reader) words will remain.<br><br>
      * 
      * It is recommended to call StringUtil.escapeCharDollarAndBackslash(text) 
@@ -345,7 +345,7 @@ public class WikiParser {
     
     
     /** Removes boundaries of something (e.g. double or triple apostrophes) 
-     * used in pairs, e.g. ''italics'' -> italics. .<br><br>
+     * used in pairs, e.g. ''italics'' -&gt; italics. .<br><br>
      * 
      * It is recommended to call StringUtil.escapeCharDollarAndBackslash(text) 
      * before this function.
@@ -374,7 +374,7 @@ public class WikiParser {
         return text;
     }
     
-    /** Removes douple apostrophes used in pairs, e.g. ''italics'' -> italics.
+    /** Removes douple apostrophes used in pairs, e.g. ''italics'' -&gt; italics.
      * It is recommended to call StringUtil.escapeCharDollarAndBackslash(text) 
      * before this function.
      */
@@ -383,7 +383,7 @@ public class WikiParser {
         return parseBounds(text, ptrn_double_apostrophe);
     }
     
-    /** Removes triple apostrophes used in pairs, e.g. '''bold''' -> bold.
+    /** Removes triple apostrophes used in pairs, e.g. '''bold''' -&gt; bold.
      * It is recommended to call StringUtil.escapeCharDollarAndBackslash(text) 
      * before this function.
      */
@@ -394,7 +394,7 @@ public class WikiParser {
     
         
     /** Removes sign of acute accent "'" for Russian wiki texts, 
-     * it is placed in the begin of article often e.g. '''itálics''' -> '''italics'''.
+     * it is placed in the begin of article often e.g. '''itálics''' -&gt; '''italics'''.
      */
     public static StringBuffer removeAcuteAccent(
             StringBuffer text,LanguageType wiki_lang)
@@ -414,20 +414,20 @@ public class WikiParser {
     /** Removes / expands interwiki, removes categories, expands wiki links.
      * 
      * @param b_remove_not_expand_iwiki if true then removes interwiki, 
-     * e.g. "[[et:Talvepalee]] text" -> " text"; else expands interwiki by 
+     * e.g. "[[et:Talvepalee]] text" -&gt; " text"; else expands interwiki by 
      * removing interwiki brackets and language code, 
-     * e.g. "[[et:Talvepalee]] text" -> "Talvepalee text".
+     * e.g. "[[et:Talvepalee]] text" -&gt; "Talvepalee text".
      *
      * @param  lang defines parsed wiki language, it is needed to remove 
      * category for the selected language, e.g. English (Category) or Esperanto 
      * (Kategorio).<br><br>
      *
      * 2. Removes categories for selected language, 
-     * e.g. English: "[[Category:Russia]] text" -> " text".<br><br>
+     * e.g. English: "[[Category:Russia]] text" -&gt; " text".<br><br>
      *
      * 3. Expands wiki links by removing brackets. There are two cases: 
-     * (1) remove brackets, e.g. [[run]] -> run and <br> 
-     * (2) [[run|running]] -> running, or [[Russian language|Russian]] -> Russian, 
+     * (1) remove brackets, e.g. [[run]] -&gt; run and <br> 
+     * (2) [[run|running]] -&gt; running, or [[Russian language|Russian]] -&gt; Russian, 
      * i.e. the visible (to reader) words will remain.
      */
     public static StringBuffer convertWikiToText(

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/XMLTag.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/XMLTag.java
@@ -19,11 +19,11 @@ import java.util.Map;
 import java.util.HashMap;
 
 
-/** XML (HTML) tags, e.g. &nbsp; &quot; &lt;br />, etc.
+/** XML (HTML) tags, e.g. &nbsp; &quot; &lt;br /&gt;, etc.
  */
 public class XMLTag {
     
-    /** Glyph character, e.g. '>', ' ', etc. */
+    /** Glyph character, e.g. '&gt;', ' ', etc. */
     private final Character glyph;
     
     /** HTML tag, e.g. '&gt;', '&nbsp;', etc. */

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/XMLTagsParser.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/text/XMLTagsParser.java
@@ -17,7 +17,7 @@ package de.tudarmstadt.ukp.jwktl.parser.ru.wikokit.base.wikipedia.text;
 
 import java.text.StringCharacterIterator;
 
-/** Parser of XML (HTML) tags, e.g. &nbsp; &quot; &lt;br />, etc.
+/** Parser of XML (HTML) tags, e.g. &nbsp; &quot; &lt;br /&gt;, etc.
  */
 public class XMLTagsParser {
     
@@ -62,7 +62,7 @@ public class XMLTagsParser {
     
     
     /** If the 'text' (from the position 'pos') is one of tags: 
-     * &lt;br />,&lt;br/>,&lt;br> then this tag is returned, 
+     * &lt;br /&gt;,&lt;br/&gt;,&lt;br&gt; then this tag is returned, 
      * else empty string will be returned.
      * 
      * @param  pos position in 'text' from which the tag will be extracted
@@ -116,11 +116,11 @@ public class XMLTagsParser {
     
     
     /** Removes the following characters from the text: 
-     * &lt;, >, &, ", also their expansions (&amp;lt;, &amp;gt;, &amp;amp;, 
+     * &lt;, &gt;, &amp;, ", also their expansions (&amp;lt;, &amp;gt;, &amp;amp;, 
      * &amp;quot;, &amp;#039;, &amp;nbsp;, &amp;ndash;, &amp;mdash;) 
      * by the 'replacement' character.<br><br>
      * 
-     * Replaces &lt;br />,&lt;br/>,&lt;br> by newline symbol.<br><br>
+     * Replaces &lt;br /&gt;,&lt;br/&gt;,&lt;br&gt; by newline symbol.<br><br>
      * 
      * Remains the character '.<br><br>
      * 
@@ -190,11 +190,11 @@ public class XMLTagsParser {
       * Escapes characters for text appearing as XML data by HTML tags.
       * 
       * <P>The following characters are replaced with corresponding character entities : 
-      * <table border='1' cellpadding='3' cellspacing='0'>
+      * <table border='1' cellpadding='3' cellspacing='0' summary="Characters to be replaced with entities">
       * <tr><th> Character </th><th> Encoding </th></tr>
-      * <tr><td> < </td><td> &amp;lt; </td></tr>
-      * <tr><td> > </td><td> &amp;gt; </td></tr>
-      * <tr><td> & </td><td> &amp;amp; </td></tr>
+      * <tr><td> &lt; </td><td> &amp;lt; </td></tr>
+      * <tr><td> &gt; </td><td> &amp;gt; </td></tr>
+      * <tr><td> &amp; </td><td> &amp;amp; </td></tr>
       * <tr><td> " </td><td> &amp;quot;</td></tr>
       * <tr><td> ' </td><td> &amp;#039;</td></tr>
       * </table>

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/util/StringUtil.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/util/StringUtil.java
@@ -130,9 +130,9 @@ public class StringUtil {
      * slash1 (true if prev prev is "\")                                
      *  slash2 (previous)                                               
      *   slash3 (current character)                                     <br>
-     *   $ -> \\$      if !slash1 && !slash2 then + "\\"                <br>
-     * \\  -> \\       if  slash1 &&  slash2 then skip                  <br>
-     * \\$ -> \\$
+     *   $ -&gt; \\$      if !slash1 &amp;&amp; !slash2 then + "\\"                <br>
+     * \\  -&gt; \\       if  slash1 &amp;&amp;  slash2 then skip                  <br>
+     * \\$ -&gt; \\$
      */
     public static StringBuffer escapeCharDollarAndBackslash(String text){
         if (null == text) {

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/util/StringUtilRegular.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikipedia/util/StringUtilRegular.java
@@ -99,7 +99,7 @@ public class StringUtilRegular {
     }
     
     /** Strips non-word letters in source array "words".
-     * E.g. {"\nword1", "\t word-long2\r\n"} -> {"word1", "word-long2"}.
+     * E.g. {"\nword1", "\t word-long2\r\n"} -&gt; {"word1", "word-long2"}.
      */
     public static void stripNonWordLetters(String words[]) {
         
@@ -128,7 +128,7 @@ public class StringUtilRegular {
     private final static String NULL_STRING = new String();
     
     /** Gets first letters till space.
-     * E.g. "word1 " -> "word1", "\t word-long2\r\n" -> "word-long2"
+     * E.g. "word1 " -&gt; "word1", "\t word-long2\r\n" -&gt; "word-long2"
      */
     public static String getLettersTillSpace(String text) {
         
@@ -140,7 +140,7 @@ public class StringUtilRegular {
     }
 
     /** Gets first letters till space " ", ... or pipe "|" (shortest string).
-     * E.g. "word1 " -> "word1", "\t word-long2\r\n" -> "word-long2"
+     * E.g. "word1 " -&gt; "word1", "\t word-long2\r\n" -&gt; "word-long2"
      * This functions is used by WPOSRu.guessPOS().
      */
     public static String getLettersTillSpaceHyphenOrPipe(String text) {
@@ -205,7 +205,7 @@ public class StringUtilRegular {
     }
 
     /** Gets first letters till first hyphen "-".
-     * E.g. "word1 " -> "word1", "\t word-long2\r\n" -> "word-long2"
+     * E.g. "word1 " -&gt; "word1", "\t word-long2\r\n" -&gt; "word-long2"
      */
     public static String getLettersTillHyphen(String text) {
 
@@ -217,7 +217,7 @@ public class StringUtilRegular {
     }
     
     
-    /** Encodes the text to latinitsa, e.g.: женьшень -> zhen'shen' (Russian) */
+    /** Encodes the text to latinitsa, e.g.: женьшень -&gt; zhen'shen' (Russian) */
     // public static String encodeRussianToLatinitsa (String text) {
     //
     // Latin1ToUTF8

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/constant/Relation.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/constant/Relation.java
@@ -111,7 +111,7 @@ public class Relation {
     }
 
     /** Gets all relations (as Array[]).
-     * Rem: We are waiting a marvellous day when JavaFX will support Collection<>.
+     * Rem: We are waiting a marvellous day when JavaFX will support Collection&lt;&gt;.
      *      Then this function will be deleted.
      */
     public static Relation[] getAllRelationsOrderedArray() {

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/en/WQuoteEn.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/en/WQuoteEn.java
@@ -21,7 +21,7 @@ public class WQuoteEn {
 
 
     /** Removes highlighted marks from a sentence.
-     * Sentence with '''words'''. -> Sentence with words.
+     * Sentence with '''words'''. -&gt; Sentence with words.
      */
     public static String removeHighlightedMarksFromSentence(String str)
     {

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/ru/WPOSRu.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/ru/WPOSRu.java
@@ -363,7 +363,7 @@ public class WPOSRu {
      *
      * {{прил ia}}
      *
-     * {{парадигма-рус  // old formatting (>500, < 1000 pages)
+     * {{парадигма-рус  // old formatting (&gt; 500, &lt; 1000 pages)
      * |шаблон=Гл11b/c
      * {{Гл1a</PRE>
      */

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/ru/WQuoteRu.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/multi/ru/WQuoteRu.java
@@ -111,8 +111,8 @@ public class WQuoteRu {
     }
 
     /** Removes highlighted marks from a sentence.
-     * 1) Sentence with '''words'''. -> Sentence with <start_replacement>words</end_replacement>.
-     * 2) Sentence with {{выдел|words}}. -> Sentence with <start_replacement>words</end_replacement>.
+     * 1) Sentence with '''words'''. -&gt; Sentence with &lt;start_replacement&gt;words&lt;/end_replacement&gt;.
+     * 2) Sentence with {{выдел|words}}. -&gt; Sentence with &lt;start_replacement&gt;words&lt;/end_replacement&gt;.
      */
     public static String removeHighlightedMarksFromSentence(String str,
                                                     String start_replacement,
@@ -132,9 +132,9 @@ public class WQuoteRu {
     }
 
     /** Additional treatment of the sentence text:
-     * 1) &nbsp;, &#160; -> " "
-     * 2) {{-}} -> " - "
-     * 3) poetry: "//" -> "\n"
+     * 1) &nbsp;, &#160; -&gt; " "
+     * 2) {{-}} -&gt; " - "
+     * 3) poetry: "//" -&gt; "\n"
      */
     public static String transformSentenceText(boolean is_sqlite, String str)
     {
@@ -156,9 +156,9 @@ public class WQuoteRu {
     }
 
     /** Additional treatment of the sentence text:
-     * 1) &nbsp;, &#160; -> " "
-     * 2) {{-}} -> "&nbsp;&mdash; " @see ru.wiktionary.org/wiki/template:-
-     * 3) poetry: "//" -> "<br>"
+     * 1) &nbsp;, &#160; -&gt; " "
+     * 2) {{-}} -&gt; "&nbsp;&mdash; " @see ru.wiktionary.org/wiki/template:-
+     * 3) poetry: "//" -&gt; "<br>"
      */
     public static String transformSentenceTextToHTML(boolean is_sqlite, String str)
     {

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/util/WikiWord.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/util/WikiWord.java
@@ -86,8 +86,8 @@ public class WikiWord {
 
     /** Removes and expands wiki links in wiki texts.<PRE>
      * Expands wiki links by removing brackets. There are two cases:
-     * (1) remove brackets, e.g. [[run]] -> run and
-     * (2) [[run|running]] -> running, or [[Russian language|Russian]] -> Russian,
+     * (1) remove brackets, e.g. [[run]] -&gt; run and
+     * (2) [[run|running]] -&gt; running, or [[Russian language|Russian]] -&gt; Russian,
      * i.e. the visible (to reader) words will remain.
      *
      * It is recommended to call StringUtil.escapeCharDollarAndBackslash(text)
@@ -137,10 +137,10 @@ public class WikiWord {
      *
      * <PRE>
      * There are cases:
-     * (1) [[run]] => two words [run, run],         remove brackets
-     * (2) [[run]]ing => two words [run, runing]    extract, remove brackets
-     * (3) [[run|running]] => two words [run, running],
-     *     or [[Russian language|Russian]] => [Russian language, Russian].
+     * (1) [[run]] =&gt; two words [run, run],         remove brackets
+     * (2) [[run]]ing =&gt; two words [run, runing]    extract, remove brackets
+     * (3) [[run|running]] =&gt; two words [run, running],
+     *     or [[Russian language|Russian]] =&gt; [Russian language, Russian].
      * </PRE>
      *
      * @param page_title    word which are described in this article 'text'

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/word/WQuote.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/ru/wikokit/base/wikt/word/WQuote.java
@@ -159,10 +159,10 @@ public class WQuote {
     }
 
     /** Removes highlighted marks from a sentence.
-     * English Wiktionary: Sentence with '''words'''. -> Sentence with words.
+     * English Wiktionary: Sentence with '''words'''. -&gt; Sentence with words.
      * Russian Wiktionary:
-     * 1) Sentence with '''words'''. -> Sentence with <start_replacement>words</end_replacement>.
-     * 2) Sentence with {{выдел|words}}. -> Sentence with <start_replacement>words</end_replacement>.
+     * 1) Sentence with '''words'''. -&gt; Sentence with &lt;start_replacement&gt;words&lt;/end_replacement&gt;.
+     * 2) Sentence with {{выдел|words}}. -&gt; Sentence with &lt;start_replacement&gt;words&lt;/end_replacement&gt;.
      */
     public static String removeHighlightedMarksFromSentence(
                                             LanguageType wikt_lang,

--- a/src/main/java/de/tudarmstadt/ukp/jwktl/parser/util/StringUtils.java
+++ b/src/main/java/de/tudarmstadt/ukp/jwktl/parser/util/StringUtils.java
@@ -77,9 +77,9 @@ public /*static*/ final class StringUtils {
 
 	/** Parses a string representation of index numbers and returns a set of
 	 *  integers containing all numbers of the specified range. The method 
-	 *  is capable of handling single numbers ("23" -> {23}), multiple comma-,
-	 *  or full-stop-separated numbers ("1,2. 4" -> {1,2,4}", and number
-	 *  ranges ("1-4" -> {1,2,3,4}). */
+	 *  is capable of handling single numbers ("23" -&gt; {23}), multiple comma-,
+	 *  or full-stop-separated numbers ("1,2. 4" -&gt; {1,2,4}", and number
+	 *  ranges ("1-4" -&gt; {1,2,3,4}). */
 	public static Set<Integer> compileIndexSet(final String indexedStr) {
 		Set<Integer> result = new TreeSet<>();
 		int lastIdx = -1;

